### PR TITLE
support multiple Insight servers usage

### PR DESCRIPTION
**Pull request by AlexanderPavlenko**

### CODE REVIEW

File: lib/straight/blockchain_adapter.rb [hunk 1]
```diff
@@ -31,24 +37,11 @@ def fetch_balance_for(address)
...
-    # @return nil
-    def self.const_missing(name)
-      Kernel.const_get(name)
-    rescue NameError
-      puts "WARNING: No blockchain adapter with the name #{name.to_s} was found!"
-      nil
+      # Converts transaction info received from the source into the
+      # unified format expected by users of BlockchainAdapter instances.
+      private def straighten_transaction(transaction)
+        raise "Please implement #straighten_transaction in #{self.to_s}"
```
Comment (AlexeyShepelev) [inline lib/straight/blockchain_adapter.rb:43]: I think, you can use `self`without `to_s`

File: lib/straight/blockchain_adapters_dispatcher.rb [hunk 2]
```diff
@@ -34,8 +34,12 @@ def execute_in_parallel(block, adapters)
       attempts_counter = Concurrent::MVar.new(adapters_to_run.size)
       pool             = Concurrent::ThreadPoolExecutor.new
       adapters_to_run.each do |adapter|
-        p = Concurrent::Promise.new(executor: pool) { block.call(adapter) }
+        p = Concurrent::Promise.new(executor: pool) {
```
Comment (AlexeyShepelev) [inline lib/straight/blockchain_adapters_dispatcher.rb:37]: You should use `do .. end` instead `{ .. }`

File: lib/straight/order.rb [hunk 3]
```diff
@@ -128,16 +128,17 @@ def get_transaction_status(reload: false, transactions: nil)
             [transaction(reload: reload)]
           end.compact
 
-        amount_paid = transactions.map { |t| t.amount }.reduce(:+) || 0
+        uniq_transactions = transactions.group_by(&:tid).each_value.map(&:first)
```
Comment (AlexeyShepelev) [inline lib/straight/order.rb]: I think you can use this construction:
`uniq_transactions = transactions.uniq(&:id)`
Isn't it?

File: lib/straight/order.rb [hunk 4]
```diff
@@ -128,16 +128,17 @@ def get_transaction_status(reload: false, transactions: nil)
...
+        amount_paid       = uniq_transactions.map { |t| t.amount }.reduce(:+) || 0
```
Comment (AlexeyShepelev) [inline lib/straight/order.rb]: `amount_paid       = uniq_transactions.map(&:amount).reduce(:+) || 0`

File: lib/straight/blockchain_adapter.rb (hunk 1 above)
Comment (AlexanderPavlenko) [inline lib/straight/blockchain_adapter.rb:43]: I can, but this is not written by me.

File: lib/straight/blockchain_adapters_dispatcher.rb (hunk 2 above)
Comment (AlexanderPavlenko) [inline lib/straight/blockchain_adapters_dispatcher.rb:37]: It depends. The second convention makes more sense to me:

> Use curly brackets for functional blocks, where the primary purpose of the block is to return a value.
> Use do...end for procedural blocks, where the primary purpose of the block is its side-effects. 

http://devblog.avdi.org/2011/07/26/the-procedurefunction-block-convention-in-ruby/

File: lib/straight/order.rb (hunk 3 above)
Comment (AlexanderPavlenko) [inline lib/straight/order.rb]: Yeah, but explicit version gives more control over which transactions are picked.
Currenly, it's a `&:first`, but who knows, maybe some additional checks will be required.

File: lib/straight/blockchain_adapters_dispatcher.rb (hunk 2 above)
Comment (AlexeyShepelev) [inline lib/straight/blockchain_adapters_dispatcher.rb:37]: May be it's good idea, but how many people read this article? How many people will ask you: why do you use this `{ .. }` for multiline code? We have the ruby style guide https://github.com/bbatsov/ruby-style-guide#user-content-single-line-blocks, let's use it.

File: lib/straight/blockchain_adapter.rb (hunk 1 above)
Comment (AlexeyShepelev) [inline lib/straight/blockchain_adapter.rb:43]: You should if you see bad code.

File: lib/straight/order.rb (hunk 3 above)
Comment (AlexeyShepelev) [inline lib/straight/order.rb]: I think it's unnecessary check. You don't use sorting, you just get first element.

**Pull request merged**